google-cloud-speech/lib/google/cloud/speech/v1/helpers.rb

```diff
@@ -86,6 +87,124 @@ def streaming_recognize streaming_config, options: nil
+          #
+          #   speech_client = Google::Cloud::Speech.new version: :v1
+          #
+          #   op = speech_client.get_operation "-"
+          #
+          #   # Process error operations.
+          #   log_error op.error if op.error?
+          #
+          #   if op.done?
+          #     # Process completed operations.
+          #     log_finished op.response, op.metadata
+          #   else
+          #     # Process pending operations.
+          #     log_pending op.name, op.metadata
+          #   end
+          #
+          def get_operation name, options: nil
+            proto_op = @operations_client.get_operation name, options: options
+            wrap_proto_op_with_gax_op proto_op, options
+          end
+
+          ##
+          # Lists operations that match the specified filter in the request. If
+          # the server doesn't support this method, it returns `UNIMPLEMENTED`.
+          #
+          # NOTE: the `name` binding below allows API services to override the
+          # binding to use different resource name schemes, such as
+          # `users/*/operations`.
+          #
+          # @param name [String]
+          #   The name of the operation collection.
+          # @param filter [String]
+          #   The standard list filter.
+          # @param page_size [Integer]
+          #   The maximum number of resources contained in the underlying API
+          #   response. If page streaming is performed per-resource, this
+          #   parameter does not affect the return value. If page streaming is
+          #   performed per-page, this determines the maximum number of
+          #   resources in a page.
+          # @param options [Google::Gax::CallOptions]
+          #   Overrides the default settings for this call, e.g, timeout,
+          #   retries, etc.
+          # @return [Google::Gax::PagedEnumerable<Google::Gax::Operation>]
+          #   An enumerable of Google::Gax::Operation instances. See
+          #   Google::Gax::PagedEnumerable documentation for other operations
+          #   such as per-page iteration or access to the response object.
+          # @raise [Google::Gax::GaxError] if the RPC is aborted.
+          # @example
+          #   require "google/cloud/speech"
+          #
+          #   speech_client = Google::Cloud::Speech.new version: :v1
+          #
+          #   # List all callbacks.
+          #   speech_client.list_operations "*", "" do |op|
+          #     # Process error operations.
+          #     log_error op.error if op.error?
+          #
+          #     if op.done?
+          #       # Process completed operations.
+          #       log_finished op.response, op.metadata
+          #     else
+          #       # Process pending operations.
+          #       log_pending op.name, op.metadata
+          #     end
+          #   end
+          #
+          def list_operations name, filter, page_size: nil, options: nil
```

Comment:
`list_operations` is not supported by the Speech API, so this will be removed before merging.

I left this method in the PR to spark a discussion on how to generalize this functionality. What do you all think about this?

Reply:
This is a good point and it's true for other operations, like `cancel` that are optional and may not be implemented by the API. 

There is currently no way for the generator to know what's implemented so if we want to generalize this well we might have to consider adding that to the configuration somehow. I wouldn't want to generate methods that don't work.

Reply:
I have updated this PR and removed `list_operations`.